src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TreeHasher.java

```diff
@@ -97,12 +101,21 @@ public void visitIdent(JCIdent tree) {
                 return;
             }
         }
-        hash(sym);
+        if (sym instanceof PoolConstant.Dynamic dynamic) {
```

Comment:
You could in principle add an `hashSymbol` method which does this dance - then you reuse it from both `visitIdent` and `visitSelect`

Reply:
I'm not sure if `visitVarDef` is ok here. Again, we seem to assume that all variables encountered here are locals - and we use a number to hash them. This is fine for locals, but I think a field should just hash to the field symbol's hash? @cushon what do you think?